### DISCUSSION TITLE
TypeError: <class 'fastapi_users_db_beanie.BeanieBaseUser'> cannot be parametrized

### DISCUSSION THREAD

**Discussion by digitros** [fastapi-users/fastapi-users#1269] [Q&A] [answered]
## Describe the bug

Below error shows when using the beanie example of the fastapi users page:
`raise TypeError(f'{cls} cannot be parametrized because it does not inherit from typing.Generic')
[1] TypeError: <class 'fastapi_users_db_beanie.BeanieBaseUser'> cannot be parametrized because it does not inherit from typing.Generic`

The code related to that is:

```
import motor.motor_asyncio
from beanie import PydanticObjectId
from fastapi_users.db import BeanieBaseUser, BeanieUserDatabase

DATABASE_URL = "mongodb://localhost:27017"
client = motor.motor_asyncio.AsyncIOMotorClient(
    DATABASE_URL, uuidRepresentation="standard"
)
db = client["database_name"]


class User(BeanieBaseUser[PydanticObjectId]):
    pass


async def get_user_db():
    yield BeanieUserDatabase(User)
```

## To Reproduce

1. Run the project

## Configuration
 - Python version :
 - FastAPI version :
 - FastAPI Users version :

### FastAPI Users configuration

```py
# Please copy/paste your FastAPI Users configuration here.
```

## Additional context

Using:
fastapi==0.100.1
uvicorn[standard]==0.23.2
fastapi-users[beanie]==12.1.0

**Comment by frankie567** [comment, marked as answer]
The code you're showing is taken from an old version. It should now look like shown here: https://fastapi-users.github.io/fastapi-users/12.1/configuration/databases/beanie/#create-the-user-model
